src/pymor/reductors/neural_network.py

```diff
@@ -504,3 +426,211 @@ def __len__(self):
         def __getitem__(self, idx):
             t = self.training_data[idx]
             return t
+
+    def train_neural_network(training_data, validation_data, neural_network,
+                             training_parameters={}):
+        """Training algorithm for artificial neural networks.
+
+        Trains a single neural network using the given training and validation data.
+
+        Parameters
+        ----------
+        training_data
+            Data to use during the training phase.
```

Comment:
Personally, I would keep it as a public function since it might be useful in different cases. Hence, I extended the documentation. Hopefully it is not too extensive now?

Reply:
Looks good! - So the function takes PyTorch data-structures as input. For a function which is part of pyMOR's public API, it would, of course, be nicer, if the inputs were pyMOR data-structures, e.g. `VectorArrays` which are then cast to PyTorch tensors. Would that make any sense here? If not, just keep the function as is ..

Reply:
Allowing for general pyMOR data-structures as input to the training method would be great. Would you be fine with checking in the training routine whether the training/validation data is already a numpy array or a PyTorch tensor and, if that is not the case, call (or at least try to call) `to_numpy` on the inputs? In def68b4 I first of all enabled the usage of numpy arrays as training data, which are fortunately already converted to PyTorch tensors automatically, if necessary, within the `DataLoader` (see [here](https://github.com/pytorch/pytorch/blob/master/torch/utils/data/_utils/collate.py#L57)).

Reply:
Yeah, allowing all three versions would be fine with me.

Reply:
The required conversion of the input data is implemented in 1890752. @sdrave Could you take a (hopefully) final look on this PR, please?

Reply:
Looking good! But you should also update the documentation ...

Reply:
Oh yes, I missed that. Done in  01a9c28.